conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/ServerRequestContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.google.common.annotations.Beta;
+import com.google.common.collect.ListMultimap;
+import java.util.List;
+import java.util.Optional;
+
+@Beta
+public interface ServerRequestContext {
```

Comment:
Alternatively: could drop the `Server` component of the name, the `com.palantir.conjure.java.undertow.lib` package name implies that to some extent.

needs class-level javadoc

Reply:
I think I prefer `RequestContext` rather than `ServerRequestContext`. The latter makes me think the lifetime of this object is attached to the the lifetime of the server rather than the request.

Reply:
Good idea.